src/features/in-app-messenger/contextual/templates/PreviewTitleBody2ButtonsPopoutTemplate.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+
+/**
+ * This component is intended to be passed to TemplateContainer as templateComponent prop, like this:
+ *   <TemplateContainer templateID='my-template' templateComponent=MyTemplate/>
+ */
+
+import * as React from 'react';
+
+import Button from '../../../../components/button/Button';
+import Overlay from '../../../../components/flyout/Overlay';
+import MessagePreviewContent from '../../../message-center/components/templates/common/MessagePreviewContent';
+import type { Token } from '../../../../common/types/core';
+
+import { type PreviewTitleBody2ButtonsPopoutParams } from '../../types';
+
+import './styles/PreviewTitleBody2ButtonsPopoutTemplate.scss';
+
+type Props = {
+    apiHost: string,
+    contentPreviewProps?: ContentPreviewProps,
+    getToken: (folderID: string | number) => Promise<Token>,
+    onAction: Function,
+    params: PreviewTitleBody2ButtonsPopoutParams,
+};
+
+const handleButton1Click = (onAction, button1) => {
+    if (button1) {
+        onAction(button1.actions);
+    }
+};
+
+const handleButton2Click = (onAction, button2) => {
+    if (button2) {
+        onAction(button2.actions);
+    }
+};
+
+const PreviewTitleBody2ButtonsPopoutTemplate = ({
+    apiHost,
+    contentPreviewProps,
+    getToken,
+    onAction,
+    params: { body, button1, button2, fileUpload: { fileId, sharedLinkUrl } = {}, title },
+}: Props) => {
+    return (
+        <div className="bdl-PreviewTitleBody2ButtonsPopoutTemplate">
+            <Overlay>
+                <div className="content-container">
+                    <div className="preview-container">
+                        <MessagePreviewContent
```

Comment:
We should move this to a common location rather than import from MC

Reply:
@DanDeMicco I opened a new PR to address this comment (had to modify many files): https://github.com/box/box-ui-elements/pull/2580